src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest.java

```diff
@@ -698,4 +699,29 @@ private static void assertRejected(Whitelist whitelist, String expectedSignature
         }
     }
 
+    @Issue("JENKINS-37129")
+    @Test public void methodMissingException() throws Exception {
+        // the case where the unsafe receiver type causes the security check to fail
+        try {
+            assertEvaluate(new GenericWhitelist(), "should fail", "[].noSuchMethod()");
+        } catch (MissingMethodException e) {
+            assertEquals(e.getType(),ArrayList.class);
+            assertThat(e.getMethod(),is("noSuchMethod"));
+        }
+
+        // trying to call an existing method that's not safe
+        try {
+            assertEvaluate(new GenericWhitelist(), "should fail", "[].class.classLoader");
+        } catch (RejectedAccessException e) {
+            assertEquals("method java.lang.Class getClassLoader", e.getSignature());
+        }
+
+        // the case where the safe receiver type causes the security check to pass
```

Comment:
Prior to the fix, I noticed that the error message was different between those two cases. `1.noSuchMethod()` already result in `MissingMethodException`, and I assumed it was because the receiver type `Integer` is already whitelisted so the sandbox invoker gets a green light to attempt to invoke a method.

`[].noSuchMethod()` failed differently, presumably because whitelist rule rejects this.